students/jongbeom/users/views.py

```diff
@@ -1,3 +1,48 @@
-from django.shortcuts import render
+import json
+import re
+
+from django.views import View
+from django.http  import JsonResponse
+
+from users.models import User
+
+class SignUpView(View):
+    def get(self, request):
+        users   = User.objects.all()
+        results = []
+
+        for user in users:
+            results.append({
+                "username"     : user.username,
+                "email"        : user.email,
+                "password"     : user.password,
+                "phone_number" : user.phone_number,
+                "created_at"   : user.created_at,
+                "updated_at"   : user.updated_at,
+            })
+        return JsonResponse({'users' : results}, status=200)
+
+    def post(self, request):
+        data = json.loads(request.body)
+        
+        try:
+            if re.match(r"^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$", data["email"]) == None:
+                    return JsonResponse({"message" : "Bad Request_Invalid Email Format"}, status=400)
+
+            if re.match(r"^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,}$", data["password"]) == None:
+                    return JsonResponse({"message" : "Bad Request_Invalid Password Format"}, status=400)
+
+            if User.objects.filter(email=data["email"]).exists():
+                    return JsonResponse({"message" : "Bad Request_Email Already Exist"}, status=400)
```

Comment:
1. 정규 표현식은 상수로 선언한 후에 match 함수에 인자로 전달해주세요-!
2. if 문에서 None으로 값을 비교하지 말고 `if re.match(r"^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$", data["email"]):`와 같이 작성해주세요. None 자체도 falsy한 값이기 때문입니다.

```suggestion
        REGEX_EMAIL = r"^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$"
        REGEX_PASSWORD = r".."
        
        try:
            if re.match(REGEX_MEAIL, data["email"]):
                    return JsonResponse({"message" : "Bad Request_Invalid Email Format"}, status=400)
            ...
```

Reply:
병민님~ `if re.match(REGEX_MEAIL, data["email"]): `부분 피드백 주셨는데 not을 붙이지 않고 반환하는 에러 값이 잘못나오는 것 같더라구요.( 이메일 중복에 대한 에러가 나와야하는 상황에서 이메일 형식에 대한 에러 메세지가 나오는 경우 처럼요.) 그래서 not 을 붙여서 수정하고 다시 동일한 http 요청을 보냈는데 제대로 출력이 되어 not을 붙여 push 했습니다! 이렇게 수정한게 맞는건가요?

위의 수정사항과 정규표현식 상수로 선언해서 다시 push 했습니다. 검토부탁드릴게요 ~!

Reply:
네 맞습니다 not을 붙여서 잘 구현해주셨습니다~~!